src/collapse/perf.rs

```diff
@@ -253,38 +253,48 @@ impl PerfState {
                 return;
             }
 
-            let mut func = with_module_fallback(module, rawfunc, pc, self.opt.include_addrs);
-            if TIDY_GENERIC {
-                func = tidy_generic(func);
-            }
+            let mut inline = vec![];
+            for func in rawfunc.split("->") {
+                let mut func = with_module_fallback(module, func, pc, self.opt.include_addrs);
+                if TIDY_GENERIC {
+                    func = tidy_generic(func);
+                }
 
-            if TIDY_JAVA && self.pname == "java" {
-                func = tidy_java(func);
-            }
+                if TIDY_JAVA && self.pname == "java" {
+                    func = tidy_java(func);
+                }
 
-            // Annotations
-            //
-            // detect kernel from the module name; eg, frames to parse include:
-            //
-            //     ffffffff8103ce3b native_safe_halt ([kernel.kallsyms])
-            //     8c3453 tcp_sendmsg (/lib/modules/4.3.0-rc1-virtual/build/vmlinux)
-            //     7d8 ipv4_conntrack_local+0x7f8f80b8 ([nf_conntrack_ipv4])
-            //
-            // detect jit from the module name; eg:
-            //
-            //     7f722d142778 Ljava/io/PrintStream;::print (/tmp/perf-19982.map)
-            if self.opt.annotate_kernel
-                && (module.starts_with('[') || module.ends_with("vmlinux"))
-                && module != "[unknown]"
-            {
-                func.push_str("_[k]");
+                // Annotations
+                //
+                // detect inlined from the inline Vec
```

Comment:
I think we should be a bit more specific here. Perhaps even link to https://github.com/brendangregg/FlameGraph/pull/89 for context.